src/Omega_h_array_ops.cpp

```diff
@@ -137,6 +137,21 @@ Write<T> divide_each(Read<T> a, Read<T> b, std::string const& name) {
   return c;
 }
 
+template <typename T>
+Write<T> concat(Read<T> a, Read<T> b, std::string const& name) {
```

Comment:
I think this may run faster as two kernels without the conditional.

Reply:
resolved here https://github.com/SCOREC/omega_h/pull/5/commits/bee665e7840480f114ea95b6ca521591bfc4fbb5